react-components/src/data/edit/EditComments.js

```diff
@@ -0,0 +1,207 @@
+            this.setState({loading : false});
+        })
+
+    }
+
+    componentWillReceiveProps(nextProps, nextContext) {
+        this.setState({open: true})
+    }
+
+    handleClose = () => {
+        this.setState({open : false});
+    };
+
+    handleSortMostRecent = () => {
+        console.log("Sort ascending");
+        let swapped;
+        let aux = 0;
+        let temp = this.state.commentList;
+        swapped = true;
+        while(swapped){
+            swapped = false;
+            for(let i = 1; i < temp.length; i++){
+                if(temp[i - 1].date > temp[i].date){
+                    swapped = true;
+                    aux = temp[i - 1];
+                    temp[i - 1] = temp[i];
+                    temp[i] = aux;
+                }
+            }
+
+        }
+        this.setState({commentList : temp});
+
+    };
+
+    handleSortLeastRecent = () => {
+        console.log("Sort descending");
+        let swapped;
+        let aux = 0;
+        let temp = this.state.commentList;
+        swapped = true;
+        while(swapped){
+            swapped = false;
+            for(let i = 1; i < temp.length; i++){
+                if(temp[i - 1].date < temp[i].date){
+                    swapped = true;
+                    aux = temp[i - 1];
+                    temp[i - 1] = temp[i];
+                    temp[i] = aux;
+                }
+            }
+
+        }
+        this.setState({commentList : temp});
+    };
+
+    handleSortOwner = () => {
+        console.log("Sort User");
+        let swapped;
+        let aux = 0;
+        let temp = this.state.commentList;
+        swapped = true;
+        while(swapped){
+            swapped = false;
+            for(let i = 1; i < temp.length; i++){
+                if(temp[i - 1].owner.charAt(0) > temp[i].owner.charAt(0)){
+                    swapped = true;
+                    aux = temp[i - 1];
+                    temp[i - 1] = temp[i];
+                    temp[i] = aux;
+                }
+            }
+
+        }
+        this.setState({commentList : temp});
+    };
+
+    handleChange = name => event => {
+        this.setState({
+            [name]: event.target.value,
+        });
+    };
+
+    createComment = () => {
+      const text = document.getElementById("addCommentTextField").value;
+      console.log(text);
+    };
+
+
+    render(){
+        const {classes} = this.props;
+        let commentItems = this.state.commentList ? this.state.commentList.map((comment, index) =>
+            <Comment message={comment.message} id={comment.id} retracted={comment.retracted}
+                    date={comment.date} owner={comment.owner}/>
+    ) : [];
+        return(
+            <div className={classes.root}>
+                <Dialog
+                    open={this.state.open}
+                    onClose={this.handleClose}
+                    id={"edit-comments-dialog"}
+                    className={classes.main}
+                >
```

Comment:
From you screenshot, the dialog looks very narrow. You can use some `maxWidth` and `fullWidth` props to make it little wider (according to the screen size). See https://material-ui.com/api/dialog/